test/unit/main_test.ts

```diff
@@ -1,7 +1,185 @@
-/// <reference path="../../typings/mocha/mocha.d.ts"/>
-/// <reference path="../../typings/chai/chai.d.ts"/>
+/// <reference path='../../typings/mocha/mocha.d.ts'/>
+/// <reference path='../../typings/chai/chai.d.ts'/>
+/// <reference path='../../typings/node/node.d.ts'/>
 
 import * as assert from 'assert';
+import * as ts from 'typescript';
+import * as chai from 'chai';
+import {Minifier, options} from '../../src/main';
 
-describe('Equality statement',
-         function() { it('shows that 1 equals 1', function() { assert.equal(1, 1); }) });
\ No newline at end of file
+function expectTranslate(code: string) {
+  var result = translateSource(code);
+  return chai.expect(result);
+}
+
+function parseFile(fileName: string, fileContent: string): ts.Program {
+  var defaultLibName = ts.getDefaultLibFilePath(options);
+  var compilerHost: ts.CompilerHost = {
+
+    getSourceFile: function(sourceName, languageVersion) {
+      return ts.createSourceFile(sourceName, fileContent, options.target, true);
+    },
+    writeFile: function(name, text, writeByteOrderMark) { var result = text; },
+    getDefaultLibFileName: () => defaultLibName,
+    useCaseSensitiveFileNames: () => false,
+    getCanonicalFileName: (filename) => filename,
+    getCurrentDirectory: () => '',
+    getNewLine: () => '\n'
+  };
+
+  var sourceFile = ts.createSourceFile(fileName, fileContent, options.target, true);
+  var program = ts.createProgram([fileName], options, compilerHost);
+
+  return program;
+}
+
+function translateSource(content: string): string {
+  var minifier = new Minifier();
+  var program = parseFile('test.ts', content);
+  var sourceFiles = program.getSourceFiles();
+  var namesToContents = {};
+
+  sourceFiles.forEach((sf) => {
+    // if (not a .d.ts file) and (is a .js or .ts file)
+    if (!sf.fileName.match(/\.d\.ts$/) && !!sf.fileName.match(/\.[jt]s$/)) {
+      namesToContents[sf.fileName] = minifier.visit(sf);
+    }
+  });
+  return namesToContents['test.ts'];
+}
+
+describe('Equality statement', () => {it('shows that 1 equals 1', () => { assert.equal(1, 1); })});
+
+describe('types', () => {
+  it('supports array types', () => {
+    expectTranslate('var list:number[];').to.equal('var list:number[];');
+    expectTranslate('var list:Array<number>;').to.equal('var list:Array<number>;');
+  });
+  it('supports enums', () => {
+    expectTranslate('enum Color {Red, Green, Blue};').to.equal('enum Color {Red, Green, Blue};');
+    expectTranslate('enum Color {Red = 1, Green, Blue};')
+        .to.equal('enum Color {Red = 1, Green, Blue};');
+  });
+  it('supports type arguments',
+     () => { expectTranslate('class X<A, B> { a: A; }').to.equal('class X<A, B> { a: A; }'); });
+});
+
+describe('functions', () => {
+  it('supports function declarations',
```

Comment:
I think you don't need so many tests - after all your code doesn't special case most of these cases. The interesting question is whether you catch all properties within these blocks - it'd be nice to have tests for two or three different syntactical structures where deep below, you find and special case a property.